server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -627,7 +649,8 @@ private String compactSegments(
       @Nullable AggregatorFactory[] metricsSpec,
       @Nullable ClientCompactionTaskTransformSpec transformSpec,
       @Nullable Boolean dropExisting,
-      @Nullable Map<String, Object> context
+      @Nullable Map<String, Object> context,
+      @Nullable ClientCompactionRunnerInfo compactionRunner
```

Comment:
This should not be `@Nullable` because we are always passing a `new ClientCompactionRunnerInfo` to it.